grails-app/controllers/org/pih/warehouse/api/CombinedShipmentItemApiController.groovy

```diff
@@ -47,12 +47,18 @@ class CombinedShipmentItemApiController {
         def vendor = Location.get(params.vendor)
         def destination = Location.get(params.destination)
         List<Order> orders = orderService.getOrdersForCombinedShipment(vendor, destination)
-        render([data: orders.findAll{ it.orderItems.any { item -> item.getQuantityRemainingToShip() > 0 } }.collect {
-            [
-                id: it.id,
-                orderNumber: it.orderNumber
-            ]
-        }] as JSON)
+        render([data: orders
+                .findAll{
+                    it.orderItems.any { item -> item.getQuantityRemainingToShip() > 0 } &&
+                    !it.orderItems.every { OrderItem item -> item.canceled}
+                }
+                .collect {
+                    [
+                        id: it.id,
+                        orderNumber: it.orderNumber
+                    ]
+                }
+        ] as JSON)
```

Comment:
I'd feel more comfortable if this was either 

``` 
    def data = ...
    render([data: data] as JSON)
```

Reply:
In fact, the code here is complex enough that it probably deserves to be migrated to the service so we can write tests for it. Since you haven't changed much about the original code (aside from formatting), we can leave it as-is, but I'd like to refactor this a bit when we get a chance.